From b54e2640a00568bf7699aa0e0caaeecf7dcc3735 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Thu, 26 Mar 2015 18:28:44 +0800
Subject: [PATCH] MLK-10475-9 ARM: dts: imx6qp-sabreauto: Add support for PRE
 and PRG

This patch adds support for PRE and PRG.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
---
 arch/arm/boot/dts/imx6qdl-sabreauto.dtsi |  2 +-
 arch/arm/boot/dts/imx6qp-sabreauto.dts   | 26 ++++++++++++++++++++++++++
 2 files changed, 27 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi b/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
index 718022d..aa261aa 100644
--- a/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
@@ -19,7 +19,7 @@
 		mxcfb3 = &mxcfb4;
 	};
 
-	memory {
+	memory: memory {
 		reg = <0x10000000 0x80000000>;
 	};
 
diff --git a/arch/arm/boot/dts/imx6qp-sabreauto.dts b/arch/arm/boot/dts/imx6qp-sabreauto.dts
index d8ac17d..a53a177 100644
--- a/arch/arm/boot/dts/imx6qp-sabreauto.dts
+++ b/arch/arm/boot/dts/imx6qp-sabreauto.dts
@@ -9,6 +9,32 @@
 #include "imx6q-sabreauto.dts"
 #include "imx6qp.dtsi"
 
+&pre1 {
+	status = "okay";
+};
+
+&pre2 {
+	status = "okay";
+};
+
+&pre3 {
+	status = "okay";
+};
+
+&pre4 {
+	status = "okay";
+};
+
+&prg1 {
+	memory-region = <&memory>;
+	status = "okay";
+};
+
+&prg2 {
+	memory-region = <&memory>;
+	status = "okay";
+};
+
 &sata {
 	status = "okay";
 };
-- 
1.8.0

